app/controllers/automated_tests_controller.rb

```diff
@@ -220,4 +222,17 @@ def required_params
   def assignment_params
     params.require(:assignment).permit(*required_params)
   end
+
+  def find_or_create_test_student(assignment_id)
```

Comment:
Think about what should happen if any of the objects don't get found or created in this method due to validation errors. What should be returned? What should be displayed to the user? Should successful changes be rolled back?